### ISSUE TITLE
Metal backend - memory leak

### ISSUE THREAD

**Issue by thevojacek** [mitsuba-renderer/nanogui#18]
When using managing GLFW on your own, using **example3** for instance, there is quite serious memory leakage when running on MacOS using Metal renderer. It is presumably due to non-releasing Metal drawables when proceeding using **example3**. There is obviously workaround to use something like `screen->drawAll()` which releases those resources and then somehow manually manage usage of system resources by thread sleeping for instance. However, this is not very convenient doing on this level, because it requires checking for incoming window events as well to determine when to start redrawing UI...

Basically what i wanted to point-out is that **example3** is unusable with Metal backend as it consumes enormous amount of memory over time and also keep requesting new ports from OS.

**Comment by wjakob** [comment]
This was fixed a while back, via the addition of an autorelease pool. If you still find this to be an issue, please let me know.
